### PR TITLE
Add setup.py and update docs to align with new purpose

### PR DESCRIPTION
**Pull request by whatnick**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,21 +24,20 @@ RUN apt-get update \
 RUN mkdir -p /code
 ADD . /code
 
-# Environment can be whatever is supported by setup.py
-# so, either deployment, test
-ARG ENVIRONMENT=deployment
-RUN echo "Environment is: $ENVIRONMENT"
-
 # Install Python requirements 
 RUN /env/bin/pip install --extra-index-url https://packages.dea.ga.gov.au/ -r /code/odc_index/requirements.txt
```
Comment (alexgleith) [inline Dockerfile]: Let's make these three a single run command, and remove the pip cache at the end.

Comment (alexgleith) [reply Dockerfile]: We should add all the requirements files up at line 17, I think.

Comment (whatnick) [reply Dockerfile]: I have moved it to a single line. Requirements are separate to allow for separate docker builds for testing vs deployment ( including pytest , moto etc. and excluding them) . Following the pattern set by explorer. I will evolve the setup.py to include runtime vs testime dependencies separately.

File: Dockerfile [hunk 2]
```diff
@@ -25,9 +25,9 @@ RUN mkdir -p /code
 ADD . /code
 
 # Install Python requirements 
-RUN /env/bin/pip install --extra-index-url https://packages.dea.ga.gov.au/ -r /code/odc_index/requirements.txt
-RUN /env/bin/pip install -r /code/tests/requirements.txt
-RUN /env/bin/pip install -r /code/assets/requirements.txt
+RUN /env/bin/pip install --extra-index-url https://packages.dea.ga.gov.au/ \
```
Comment (alexgleith) [inline Dockerfile]: I think you might want to use the `env-build-tool` in `extent` mode here, right @Kirill888? 

https://github.com/opendatacube/geobase/blob/master/base/wheels/env-build-tool

**Pull request merged**